### ISSUE TITLE
model- bag_attention.py-98行

### ISSUE THREAD

**Issue by fengyunzaidushi** [thunlp/OpenNRE#374]
if bag_size > 0:
            token = token.view(-1, token.size(-1))
            pos1 = pos1.view(-1, pos1.size(-1))
            pos2 = pos2.view(-1, pos2.size(-1))
            if mask is not None:
                mask = mask.view(-1, mask.size(-1))
        else:
            begin, end = scope[0][0], scope[-1][1]
            token = token[:, begin:end, :].view(-1, token.size(-1))
            pos1 = pos1[:, begin:end, :].view(-1, pos1.size(-1))
            pos2 = pos2[:, begin:end, :].view(-1, pos2.size(-1))
            if mask is not None:
                mask = mask[:, begin:end, :].view(-1, mask.size(-1)
            scope = torch.sub(scope, torch.zeros_like(scope).fill_(begin))

请问 这句代码是什么作用： scope = torch.sub(scope, torch.zeros_like(scope).fill_(begin))
谢谢！

**Comment by demerzel-iv** [comment]
由于这里的93-97行对token,pos1,pos2,mask进行了切片，因此切片之后新的scope也应当向左平移对应的长度

**Comment by fengyunzaidushi** [comment]
> 由于这里的93-97行对token,pos1,pos2,mask进行了切片，因此切片之后新的scope也应当向左平移对应的长度

谢谢！
